src/SharedModule/components/SideBar/SideBar.tsx

```diff
@@ -1,7 +1,56 @@
-import React from 'react'
+import { Sidebar, Menu, MenuItem} from "react-pro-sidebar";
+import { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import ChangPass from '../../../AuthModule/components/ChangPass/ChangPass';
+import  {Modal}  from "react-bootstrap";
 
 export default function SideBar() {
+  const navigate = useNavigate();
+  const [isCollapsed, setIsCollapsed] = useState(true);
+  const toggleCollapse = () => {
+    setIsCollapsed(!isCollapsed);
+  };
+  // closing and opening Modal
+  const [show, setShow] = useState<boolean>(false);
+  const handleClose = () => {
+    setShow(false);
+  };
+  const handleShow = () => setShow(true);
+  // function Logout 
+  const logOut = () => {
+    localStorage.removeItem("adminToken");
+    localStorage.removeItem("token");
+    navigate("/login")
+  }
   return (
-    <div>SideBar</div>
-  )
-}
+    <>
+    <Modal  show={show} onHide={handleClose}>
+        <Modal.Header closeButton></Modal.Header>
+        <Modal.Body>
+          <ChangPass handleClose={handleClose} />
+        </Modal.Body>
+      </Modal>
+      <div className="sidbar-con">
+      <Sidebar  collapsed={isCollapsed}>
+        <Menu>
+          <div className="position-relative mb-4 pt-2"  onClick={toggleCollapse}>
```

Comment:
This should be a button.

Reply:
There is an issue also with the sidebar toggles, you'll notice that we can only toggle it by clicking on the icon, but the outer part is not clickable